pkg/dashboard/apiserver/common/common.go

```diff
@@ -425,6 +436,144 @@ func (s *Service) getRbacConfig(c *gin.Context) {
 	c.JSON(http.StatusOK, rbacMap)
 }
 
+// @Summary Get physicalMachines from Kubernetes cluster.
+// @Description Get physicalMachines from Kubernetes cluster.
+// @Tags common
+// @Produce json
+// @Param request body v1alpha1.PhysicalMachineSelectorSpec true "Request body"
+// @Success 200 {array} PhysicalMachine
+// @Router /common/physicalmachines [post]
+// @Failure 500 {object} utils.APIError
+func (s *Service) listPhysicalMachines(c *gin.Context) {
+	kubeCli, err := clientpool.ExtractTokenAndGetClient(c.Request.Header)
+	if err != nil {
+		utils.SetAPIError(c, utils.ErrBadRequest.WrapWithNoMessage(err))
+		return
+	}
+
+	selector := v1alpha1.PhysicalMachineSelectorSpec{}
+	if err := c.ShouldBindJSON(&selector); err != nil {
+		utils.SetAPIError(c, utils.ErrBadRequest.WrapWithNoMessage(err))
+		return
+	}
+	ctx := context.TODO()
+	filtered, err := physicalmachine.SelectPhysicalMachines(ctx, kubeCli, nil, selector, s.conf.ClusterScoped, s.conf.TargetNamespace, s.conf.EnableFilterNamespace)
+	if err != nil {
+		utils.SetAPIError(c, utils.ErrInternalServer.WrapWithNoMessage(err))
+		return
+	}
+
+	physicalMachines := make([]PhysicalMachine, 0, len(filtered))
+	for _, pm := range filtered {
+		physicalMachines = append(physicalMachines, PhysicalMachine{
+			Name:      pm.Name,
+			Namespace: pm.Namespace,
+			Address:   pm.Spec.Address,
+		})
+	}
+
+	c.JSON(http.StatusOK, physicalMachines)
+}
+
+// @Summary Get the labels of the physicalMachines in the specified namespace from Kubernetes cluster.
+// @Description Get the labels of the physicalMachines in the specified namespace from Kubernetes cluster.
+// @Tags common
+// @Produce json
+// @Param physicalMachineNamespaceList query string true "The physicalMachine's namespace list, split by ,"
+// @Success 200 {object} MapSlice
+// @Router /common/physicalmachine-labels [get]
+// @Failure 500 {object} utils.APIError
+func (s *Service) getPhysicalMachineLabels(c *gin.Context) {
+
+	kubeCli, err := clientpool.ExtractTokenAndGetClient(c.Request.Header)
+	if err != nil {
+		utils.SetAPIError(c, utils.ErrBadRequest.WrapWithNoMessage(err))
+		return
+	}
+
+	labels := make(map[string][]string)
```

Comment:
how about moving the definition of `labels` before the place use it?

Reply:
done